### ISSUE TITLE
Importing NSC generated ops and users and accounts

### ISSUE THREAD

**Issue by aKardasz** [edgefarm/vault-plugin-secrets-nats#19]
Hello,

What would the order of imports be if I wanted to import all the data created previously by the nsc tool?

I am having a bit of trouble adding everything in to be able to keep generating accounts and users, while updating the nats servers with this new setup.

**Comment by batthebee** [comment]
Hi,

Normally, the order should not matter. What exactly is not working? Do you get meaningful logs?

**Comment by aKardasz** [comment]
So I originally imported as follows:

Operator Nkey,
Operator Signing Key

sys account nkey
sys account signing key

etc, but this didn't show up as an ops, acc, or user. So then using those keys I created ops, and acc, user using issue, however none of them linked, they created successfully though, however I think this may have been due to how I was running each command and what I was passing in. 

What we are trying to do is bootstrap the initial set of credentials and then pass all this bootstrapped info to the vault instance that will actually be generating the credentials. (It would have to be done this way due to how the environment is setup currently)

So just trying to get the initial ops, sys and sys user made, export these and import into the new vault instance and continue on from there. Also want to thank you guys for all the hard work put into this plugin.

**Comment by aKardasz** [comment]
Update: So I was able to get a set of credentials created using the signing keys etc ( account and user ), and when I manually added them to nsc, they worked, however for some reason the plugin keeps giving me unauthorized when trying to sync. On the nats side it is saying that the jwt does not exist, however I copied them directly from the config file and imported them manually into vault. Is there any info I can provide to possibly see if I can get passed this sync issue. ( the operators, sys account, and sys user where created previously using nsc, and then imported into the plugin.

**Comment by aKardasz** [comment]
I was able to get vault to spit out: "cannot sync account server: system account user nkey does not exist"

However, the nkey is there, and was manually imported

**Comment by batthebee** [comment]
Thanks for the infos, we will check that!

**Comment by aKardasz** [comment]
Yeah I figured it out, you pretty much have to generate the various pieces in a certain order for it to all pick up nicely. Might be worth looking into a feature for importing from nsc or a small script to do so. (I will probably get around to something like this, just have to finish a few other items on my list of todo's). On another note, is there anyway to force sync vault to the nats server, more so if I want it to sync right after a user/credential is made so that there is no 2 minute delay between creation and being able to utilize.

**Comment by aKardasz** [comment]
@batthebee wow wasn't expecting such a quick turn around. Thank you! 

I think I may have found another bug. 

When the plugin is running and in sync mode and some one creates a set of credentials and pushes them with nsc cli the nats vault plugin seems to not be able to connect to the server. 


`{"level":"error","time":1701991369,"message":"resolver: failed to get response to create: nats: timeout"}`

` {"level":"error","operator":"<OPERATOR_NAME>","account":"<ACCOUNT_NAME>","error":"no response from server","time":1701991369,"message":"cannot sync account server (add)"}`

This only occurs after someone pushes up credentials with nsc cli. A restart of vault seems to fix the issue.
